kratos/geometries/tetrahedra_3d_10.h

```diff
@@ -621,6 +621,20 @@ template<class TPointType> class Tetrahedra3D10 : public Geometry<TPointType>
         return faces;
     }
 
+    /** This method calculates and returns the average edge length of the geometry
+     *
+     * @return double value with the average edge length
+     *
+     */
+    double AverageEdgeLength() const override {
+        double average_edge_length = 0.0;
+        GeometriesArrayType edges = this->GenerateEdges();
+        for (auto& edge : edges) {
+            average_edge_length += edge.Length();
+        }
+        return average_edge_length/6.0;
```

Comment:
```suggestion
        const GeometriesArrayType edges = this->GenerateEdges();
        return std::accumulate(
            edges.begin(),
            edges.end(),
            0.0,
            [](double sum, const auto& rEdge) {return sum + rEdge.Length();}
        );
```

Reply:
done, i dont see clearly the advantage though, is it just it looks better?

Reply:
In this case yes, it probably makes no practical difference because they're so few items in `edges`, but it's more straightforward to read what's happening.

In the general case though, it pays off to use standard algorithms. They're usually implemented with exception guarantees and in a manner that makes compiler optimizations possible. Also, they usually switch between algorithms depending on the input size to provide better performance / stable numerics.

Reply:
I guess you need to `#include <numeric>`

btw I haven't compiled and tested the suggestion, I meant it as a comment.

Reply:
yep, I will add it now